### PR TITLE
[Arm64] SIMDIntrinsicGetItem

### PR DESCRIPTION
**Pull request by sdmaclea**
Fix srcCount
Fix item containment
Implement [Arm64] genSIMDIntrinsicGetItem w/ non const index
Handle out of range const index
~~Handle SIMD12 special case~~

@dotnet/jit-contrib @dotnet/arm64-contrib PTAL

### CODE REVIEW

File: src/jit/lowerarmarch.cpp [hunk 1]
```diff
@@ -777,7 +777,15 @@ void Lowering::ContainCheckSIMD(GenTreeSIMD* simdNode)
...
             // If the index is a constant, mark it as contained.
-            CheckImmedAndMakeContained(simdNode, op2);
+            if (op2->IsCnsIntOrI())
+            {
+                MakeSrcContained(simdNode, op2);
+            }
+            else
+            {
+                // If the index is not a constant, we will use the SIMD temp location to store the vector.
+                comp->getSIMDInitTempVarNum();
```
Comment (BruceForstall) [inline src/jit/lowerarmarch.cpp]: I notice that xarch calls `getSIMDInitTempVarNum` in TreeNodeInfoInitSIMD. Is there a reason why you call it here instead?

File: src/jit/codegenarm64.cpp [hunk 2]
```diff
@@ -4499,19 +4507,43 @@ void CodeGen::genSIMDIntrinsicGetItem(GenTreeSIMD* simdNode)
...
 
+    emitAttr baseTypeSize = emitTypeSize(baseType);
+
     if (op2->IsCnsIntOrI())
     {
         assert(op2->isContained());
 
-        emitAttr     attr  = emitTypeSize(baseType);
-        unsigned int index = (unsigned int)op2->gtIntCon.gtIconVal;
+        int index = (unsigned int)op2->gtIntCon.gtIconVal;
```
Comment (BruceForstall) [inline src/jit/codegenarm64.cpp]: Can you also change the cast to `(int)`?

File: src/jit/codegenarm64.cpp [hunk 3]
```diff
@@ -4499,19 +4507,43 @@ void CodeGen::genSIMDIntrinsicGetItem(GenTreeSIMD* simdNode)
...
+
+        unsigned baseTypeScale = genLog2(EA_SIZE_IN_BYTES(baseTypeSize));
+
+        // Load the address of simdInitTempVarNum
+        getEmitter()->emitIns_R_S(INS_lea, EA_PTRSIZE, tmpReg, simdInitTempVarNum, 0);
+
+        // Store the vector to simdInitTempVarNum
+        getEmitter()->emitIns_R_R(INS_str, emitTypeSize(simdType), srcReg, tmpReg);
+
+        // Load item at simdInitTempVarNum[index
```
Comment (BruceForstall) [inline src/jit/codegenarm64.cpp]: missing trailing `]`

File: src/jit/lowerarmarch.cpp (hunk 1 above)
Comment (sdmaclea) [reply src/jit/lowerarmarch.cpp]: I can move it.  At the time it felt like a better place.  At the moment I see no reason for my initial opinion.

**Comment by sdmaclea** [comment]
@BruceForstall I addressed your issues (force pushed).

**Pull request merged**